components/gcp-click-to-deploy/src/configs/cluster-kubeflow-util.jinja

```diff
@@ -0,0 +1,27 @@
+{% set SECRETS_COLLECTION = '/api/v1/namespaces/{namespace}/secrets' %}
+{% set CLUSTER_TYPE = env['project'] + '/' + properties["clusterType"] %}
+
+resources:
+- name: kubeflow-admin-key
+  type: {{ CLUSTER_TYPE }}:{{ SECRETS_COLLECTION }}
+  properties:
+    apiVersion: v1
+    kind: Secret
+    namespace: {{ properties["namespace"] }}
+    metadata:
+      name: admin-gcp-sa
+    type: Opaque
+    data:
+      admin-gcp-sa.json: {{ properties["saKey"] }}
```

Comment:
This is a smart idea; using DM to insert the service account key into the cluster. Unfortunately this means the secret key would be stored in DM and viewable by anyone with access to the project. I don't think we want that.

An alternative approach would be to just use the service account as the service account for the GKE nodes.
The downside of that approach would be that it means all pods on that node would run with elevated privileges which would be undesirable.

Its possible that we could use RBAC to restrict what can run on a specific node pool so that other processes couldn't run on that node pool.

Reply:
Currently our code assume those keys will be stored in k8s cluster, so anyone with project access can view from cluster secrets?
Another way is maybe we can expire those keys once we finish deploy?

Reply:
By using VM account for nodes, we should be able to avoid exposing secrets.
Handled in separate PR: https://github.com/kubeflow/kubeflow/pull/1302

Reply:
I think putting them in DM might be an enhanced risk because K8s secrets are namespace scoped resource so not everyone might have access to the namespace. I'm not even sure project viewers do.